core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java

```diff
@@ -1398,4 +1398,37 @@ private static Double mapL2Norm(Block map)
 
         return Math.sqrt(norm);
     }
+
+    @Description("Black-Scholes equation for a Euro vanilla option")
+    @ScalarFunction("bs_european_option")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double bsEuropeanOption(
+            @SqlType(StandardTypes.DOUBLE) double stockPrice,
+            @SqlType(StandardTypes.DOUBLE) double strikePrice,
+            @SqlType(StandardTypes.DOUBLE) double timeToExpiration,
+            @SqlType(StandardTypes.DOUBLE) double riskFreeRate,
+            @SqlType(StandardTypes.DOUBLE) double volatility,
+            @SqlType("varchar(x)") String optionType)
```

Comment:
I would probably be better to have two functions: `black_scholes_call` and `black_scholes_put`. Simulating an enum with a varchar argument is going to make the computation unnecessarily more expensive.